clients/cobol-lsp-vscode-extension/src/test/suite/processorgroups.spec.test.ts

```diff
@@ -0,0 +1,119 @@
+}
+
+function remove_pgm_cfg(pgmConf: PgmConf) {
+  const pgm_conf_path = path.join(getWorkspacePath(), ".cobolplugin", "pgm_conf.json");
+  const fileContents = fs.readFileSync(pgm_conf_path, 'utf8');
+  const pgmConfs = JSON.parse(fileContents);
+  pgmConfs.pgms = pgmConfs.pgms.filter(
+    (obj: PgmConf) => obj.program !== pgmConf.program && obj.pgroup !== pgmConf.pgroup
+  );
+  const updatedJson = JSON.stringify(pgmConfs, null, 4);
+  fs.writeFileSync(pgm_conf_path, updatedJson, 'utf8');
+}
+
+function add_pgm_cfg(pgmConf: PgmConf) {
+  const pgm_conf_path = path.join(getWorkspacePath(), ".cobolplugin", "pgm_conf.json");
+  const fileContents = fs.readFileSync(pgm_conf_path, 'utf8');
+  const pgmConfs = JSON.parse(fileContents);
+  pgmConfs.pgms.push(pgmConf);
+  const updatedJson = JSON.stringify(pgmConfs, null, 4);
+  fs.writeFileSync(pgm_conf_path, updatedJson, 'utf8');
+}
+
+suite("Integration Test Suite: Processor Groups", function () {
+  suiteSetup(async function () {
+    this.timeout(helper.TEST_TIMEOUT);
+    helper.updateConfig("basic.json");
+    await helper.activate();
+  });
+
+  const idmsPgmConf = { "program": "cobol-idms/*", "pgroup": "IDMS" };
+  const dacoPgmConf = { "program": "cobol-daco/*", "pgroup": "DaCo" };
+
+  this.afterEach(async () => await helper.closeAllEditors()).timeout(
+    helper.TEST_TIMEOUT,
+  );
+
+  // cobol-dialect-api (development) npm install && npm run compile
+  // daco-dialect-support (development) npm install && npm run package
+  // idms-dialect-support (development) npm install && npm run package
+  // put server.jar, dialect-idms.jar and dialect-daco.jar into their server/jar folder
+
+  test("TC355920: IDMS - processor group", async () => {
+    add_pgm_cfg(idmsPgmConf)
+    const extSrcPath = path.join("cobol-idms", "DACRFI11N.cbl");
+    await helper.showDocument(extSrcPath);
+    const editor = helper.get_editor(extSrcPath);
+    const diagnostics = vscode.languages.getDiagnostics(editor.document.uri);
+    assert.strictEqual(diagnostics.length, 0);
+    await helper.closeAllEditors();
+  })
+    .timeout(helper.TEST_TIMEOUT)
+    .slow(1000);
+  
+  test("TC355921: DaCo - processor group", async () => {
+    add_pgm_cfg(dacoPgmConf)
+    const extSrcPath = path.join("cobol-daco", "DABPA11N.cbl");
+    await helper.showDocument(extSrcPath);
+    const editor = helper.get_editor(extSrcPath);
+    const diagnostics = vscode.languages.getDiagnostics(editor.document.uri);
+    assert.strictEqual(diagnostics.length, 0);
+    await helper.closeAllEditors();
+  })
+    .timeout(helper.TEST_TIMEOUT)
+    .slow(1000);
+
+  test("TC355918: IDMS - No processor group", async () => {
+    remove_pgm_cfg(idmsPgmConf)
+    const extSrcPath = path.join("cobol-idms", "DACRFI11N.cbl");
+    await helper.showDocument(extSrcPath);
+    const editor = helper.get_editor(extSrcPath);
+    await helper.waitFor(
+      () => vscode.languages.getDiagnostics(editor.document.uri).length == 1,
+    );
```

Comment:
Shouldn't the waitFor throw or something when it times out? Or at least return true/false?